modules/neovim/init/basic.nix

```diff
@@ -158,6 +158,35 @@ in {
       default = "sensitive";
       description = "Set the case sensitivity of search";
     };
+
+    undoFile = {
+      enable = mkEnableOption ''
+        undofile for Neovim.
+
+        ::: {.warning}
+        [{option}`vim.undoFile.path`](#opt-vim.undoFile.path) **must** be set for this
+        option to have any effect. If unset, undofile options will be ignored altogether.
+        :::
```

Comment:
This would cause an error if `enable` is true but a path is not set explicitly. `null` default is a sanity check which is also used in optionalString.